enhancements/api-review/add-repositoryMirrors-spec.md

```diff
@@ -0,0 +1,175 @@
+
+### Goals
+
+- Enable the user to pull images through ImageContentSourcePolicy configured mirrors by tags.
+
+### Non-Goals
+
+## Proposal
+
+The Image API has an optional `MirrorByDigestOnly` filed with type `bool` to `Registry`.
+
+```go
+// Registry represents a registry.
+type Registry struct {
+  // ...
+
+	// If true, mirrors will only be used for digest pulls. Pulling images by
+	// tag can potentially yield different images, depending on which endpoint
+	// we pull from.  Forcing digest-pulls for mirrors avoids that issue.
+	MirrorByDigestOnly bool `toml:"mirror-by-digest-only,omitempty"`
+}
+```
+
+The ImageContentSourcePolicy CRD has the cluster-wide information about handle the mirrors only when pulling the images that are referenced by their digests.
+It will now watch for mirrors configured through spec `repositoryMirrors` to not require digest reference.
+
+An example ImageContentSourcePolicy file will look like:
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryMirrors:
+  - mirrors:
+    - example.io/example/ubi-minimal 
+    source: registry.access.redhat.com/ubi8/ubi-minimal
+```
+
+### User Stories
+
+#### As a user, I would like to pull images from mirror using tag, without digests reference
+The user need to define multiple ImageContentSourcePolicies, used by apps/manifests that don't use digests to pull
+the images when working with disconnected environments or pulling from registries that act as transparent pull-through proxy cache.
+The user can pull image without digest by configuring `repositoryMirrors` spec in the ImageContentSourcePolicy file.
+And create the ImageContentSourcePolicy project. Once this is done, the images can be pulled from the mirrors without the digest referenced.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in:
+- openshift/api
+- openshift/machine-config-operator
+
+This is an example of the ImageContentSourcePolicy file:
+
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryMirrors:
+  - mirrors:
+    - example.io/example/ubi-minimal 
+    source: registry.access.redhat.com/ubi8/ubi-minimal 
+```
+
+The ImageContentSourcePolicy file will create a drop-in file at `/etc/containers/registries.conf` that looks like:
+
+```toml
+unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
+[[registry]]
+  location = "registry.access.redhat.com/ubi8/"
+  insecure = false
+  blocked = false
+  # No mirror-by-digest-only = true configured
+  prefix = ""
+
+  [[registry.mirror]]
+    location = "example.io/example/ubi8-minimal"
+    insecure = false
+```
+
+If both `repositoryMirrors` and `repositoryDigestMirrors` are configured for same “source” repository, the ImageContentSourcePolicy will be rejected.
```

Comment:
With a property for digest-ness under `repositoryMirrors`, maybe time to go v1alpha2 and drop `repositoryDigestMirrors`?

Reply:
If the repositoryDigestMirrors dropped, the digest requirement will not exist. Is it possible the customer still wants to limit the use of mirrors? This looks like a very rare case, I am not sure.

Reply:
`mirror-by-digest-only: true` is sane, and protects you from malicious/compromised registries by forcing you to use by-digest pullspecs.  But you can ask for one if those with:

```yaml
apiVersion: operator.openshift.io/v1alpha1
kind: ImageContentSourcePolicy
metadata:
  name: ubi8repo
spec:
  repositoryMirrors:
  - mirrors:
    - example.io/example/ubi-minimal 
    source: registry.access.redhat.com/ubi8/ubi-minimal
  allowMirrorByTags: false
```

without needing a `repositoryDigestMirrors` entry.

Reply:
> With a property for digest-ness under `repositoryMirrors`, maybe time to go v1alpha2 and drop `repositoryDigestMirrors`?

@wking  How we do the API version change and the drop for `repositoryDigestMirrors`? Do we add the new `repositoryMirrors` and drop the old one at the same time in openshift/api?

Should this enhancement mention we are going to drop the `repositoryDigestMirrors`?